src/main/java/in/koreatech/koin/domain/graduation/service/GraduationService.java

```diff
@@ -0,0 +1,176 @@
+                if (row.getRowNum() == 0 || skipRow(data))
+                    continue;
+
+                if (data.excelClassTitle().equals(TOTAL))
+                    break;
+
+                String semester = getKoinSemester(data.excelSemester(), data.excelYear());
+                CourseType courseType = courseTypeRepository.findByName(data.excelCourseType()).orElse(null);
+                Lecture lecture = lectureRepositoryV2.findBySemesterAndCodeAndLectureClass(semester,
+                    data.excelCode(), data.excelLectureClass()).orElse(null);
+
+                if (!currentSemester.equals(semester)) {
+                    currentSemester = semester;
+                    graduationFrame = createFrameAboutExcel(userId, currentSemester);
+                }
+
+                TimetableLecture timetableLecture = TimetableLecture.builder()
+                    .classTitle(data.excelClassTitle())
+                    .classTime(lecture != null ? lecture.getClassTime() : null)
+                    .classPlace(null)
+                    .professor(data.excelProfessor())
+                    .grades(data.excelCredit())
+                    .memo(null)
+                    .isDeleted(false)
+                    .lecture(lecture)
+                    .timetableFrame(graduationFrame)
+                    .courseType(courseType)
+                    .build();
+
+                timetableLectureRepositoryV2.save(timetableLecture);
+            }
+        }
+    }
+
+    private TimetableFrame createFrameAboutExcel(Integer userId, String semester) {
+        User user = userRepository.getById(userId);
+        Semester saveSemester = semesterRepositoryV2.getBySemester(semester);
+
+        List<TimetableFrame> timetableFrameList = timetableFrameRepositoryV2.findAllByUserIdAndSemesterId
+            (userId, saveSemester.getId());
+        for (TimetableFrame timetableFrame : timetableFrameList) {
+            if (timetableFrame.isMain()) {
+                timetableFrame.cancelMain();
+            }
+        }
+        TimetableFrame graduationFrame = TimetableFrame.builder()
+            .user(user)
+            .semester(saveSemester)
+            .name("졸업학점 계산용 테이블")
+            .isDeleted(false)
+            .isMain(true)
+            .build();
+        timetableFrameRepositoryV2.save(graduationFrame);
+
+        return graduationFrame;
+    }
+
+    private void checkFiletype(MultipartFile file) {
+        if (file == null) {
+            throw new ExcelFileNotFoundException("파일이 있는지 확인 해주세요.");
+        }
+
+        String fileName = file.getOriginalFilename();
+        int findDot = fileName.lastIndexOf(".");
+        if (findDot == -1) {
+            throw new ExcelFileNotFoundException("파일의 형식이 맞는지 확인 해주세요.");
+        }
+
+        String extension = fileName.substring(findDot + 1);
+        if (!extension.equals("xls") && !extension.equals("xlsx")) {
+            throw new ExcelFileCheckException("엑셀 파일인지 확인 해주세요.");
```

Comment:
## R
어..! 이렇게 되면 `.xls`도 막히는 거 아닌가용..?

Reply:
아아 착각했네욥,,

Reply:
뭔가 가독성 더 좋아보이는 코드 투척
```suggestion
        if (!(extension.equals("xls") || extension.equals("xlsx"))) {
            throw new ExcelFileCheckException("엑셀 파일인지 확인 해주세요.");
```